### PR TITLE
feat: command-line tool request definition

### PR DESCRIPTION
**Pull request by sqin2019**

### CODE REVIEW

File: apis/v1alpha1/cli_request.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+// CLIRequest represents a request to run CLI commands.
+type CLIRequest struct {
+	// CLI name such as gcloud.
+	CLI string `yaml:"cli,omitempty"`
```
Comment (sqin2019) [inline apis/v1alpha1/cli_request.go:20]: This is here because we may want to support other CLI like aws.

**Comment by capri-xiyue** [comment]
Do you need to leverage this https://github.com/abcxyz/pkg/tree/main/cli when we implement the AOD cli?

**Comment by sqin2019** [comment]
> Do you need to leverage this https://github.com/abcxyz/pkg/tree/main/cli when we implement the AOD cli?

Yes, for example,  `aod cli do -path "/path/to/<CLIRequestFileName>.yaml"` to handle CLIRequest.

**Comment by capri-xiyue** [comment]
> > Do you need to leverage this https://github.com/abcxyz/pkg/tree/main/cli when we implement the AOD cli?
> 
> Yes, for example, `aod cli do -path "/path/to/<CLIRequestFileName>.yaml"` to handle CLIRequest.

If we use abcxyz/pkg/cli
do you still need to define such type CLIRequest struct? I thought the CLI request here should look like https://github.com/abcxyz/pmap/blob/main/pkg/cli/mapping_validate.go#L32 using abcxyz/pkg/cli

**Review by capri-xiyue — APPROVED** [review]

**Pull request merged**